File: src/led/LEDStripRenderer.java
package led;

import java.awt.Color;

import javax.swing.JLabel;

public class LEDStripRenderer {

	private Color[] colorData = new Color[300];
	public static JLabel[] leds;
	
	
	
	public void setPixel(int _index, Color _color) {
		colorData[_index] = _color;
	}
	
	public void render() {
		for (int i = 0; i < leds.length; i++) {
			leds[i].setForeground(colorData[i]);
		}
	}
}

File: src/main/java/led/LEDStripRenderer.kt
package led

import java.awt.Color
import javax.swing.JLabel

class LEDStripRenderer {
    private val colorData = arrayOfNulls<Color>(300)
    fun setPixel(_index: Int, _color: Color?) {
        colorData[_index] = _color
    }

    fun render() {
        for (i in leds.indices) {
            leds[i]!!.foreground = colorData[i]
        }
    }

    companion object {
        lateinit var leds: Array<JLabel?>
    }
}